xpass/xpass.cc

```diff
@@ -229,6 +269,25 @@ void XPassAgent::handle_retransmit() {
   }
 }
 
+void XPassAgent::handle_retransmit_credit_stop() {
+  switch (credit_recv_state_){
+    case XPASS_RECV_CREDIT_STOP_SENT:
+    credit_recv_state_ = XPASS_RECV_CLOSE_WAIT;
+      break;
+  case XPASS_RECV_CLOSE_WAIT:
+    if (credit_recv_count_ == 0) {
+      credit_recv_state_ = XPASS_RECV_CLOSED;
+    retransmit_credit_stop_timer_.force_cancel();
+    return;
+    }
+    // retransmit credit_stop
+    send(construct_credit_stop(), 0);
+    break;
+  }
+  credit_recv_count_ = 0;
```

Comment:
For consistency (with credit_total_, credit_dropped_), let's change this to "credit_recved_"